torch/autograd/gradcheck.py

```diff
@@ -411,8 +411,8 @@ def fn(input):
                     return fail_test('Numerical gradient for function expected to be zero')
         return True
 
-    for i, o in enumerate(output):
-        if not o.requires_grad:
+    for i, o in enumerate(func_out):
```

Comment:
Why change this? The filtering above should be already removing all the non-Tensor outputs?

Reply:
I am not sure about these changes.
Can you add some tests to make sure it does the right thing if you keep these?